### ISSUE TITLE
feat(`config`): include desired `foundry` version in `foundry.toml`

### ISSUE THREAD

**Issue by iainnash** [foundry-rs/foundry#6084]
### Component

Forge, Foundryup

### Describe the feature you would like

Support setting the foundry version for foundryup in the foundry.toml file.

This allows teams and projects to "pin" the correct version of foundry that known works with their repository across all CI tools and teammates and helps debugging errors with builds and verification.

Either a warning can be shown for a wrong version or foundryup can automatically switch to the necessary version similar to how "engines" works within package.json or how nvm works in the javascript realm.

**Comment by zerosnacks** [comment]
Blocked by: https://github.com/foundry-rs/foundry/issues/3895

See https://github.com/foundry-rs/foundry/labels/A-releases for other related tickets

**Comment by zerosnacks** [comment]
Closing in favor of https://github.com/foundry-rs/foundry/issues/5133, marking as duplicate
